### PR TITLE
Fix tests after diff bump; drive-by prettify diff tests

### PR DESCRIPTION
**Pull request by kyegupov**

### CODE REVIEW

**Comment by maghoff** [comment]
Sweet, thanks! I'll have to look a bit closer at it before merging to verify that this is the result I want, since this actually changes the tests.

**Comment by kyegupov** [comment]
Yeah, the question is whether you want to preserve a trailing newline from the input or not.
I'd vote yes, because otherwise you will be losing information, and that is better done later than earlier.

File: src/merge/mod.rs [hunk 1]
```diff
@@ -95,7 +98,7 @@ pub fn merge_lines<'a>(a: &'a str, o: &'a str, b: &'a str) -> MergeResult<&'a st
                     Resolved(y) => y.into_iter(),
                     _ => unreachable!()
                 })
-                .flat_map(|x| vec![x, "\n"].into_iter())
+                .intersperse("\n")
```
Comment (maghoff) [inline src/merge/mod.rs]: While this is an improvement, I would prefer not to add an additional dependency for this single line. Can you revert this, please? :)

File: src/merge/mod.rs [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+
 mod chunk_iterator;
 mod chunk;
 mod output;
 
 use std::fmt::Debug;
 
-use diff;
 
+use diff;
```
Comment (maghoff) [inline src/merge/mod.rs]: The `use`s are grouped: imports from `std` first, imports from external crates and imports from other modules. Please retain this grouping (ie. revert this change)

**Comment by maghoff** [comment]
This is good! I'm ready to merge this when my above comments have been addressed :) Thanks again.

File: src/merge/mod.rs (hunk 1 above)
Comment (kyegupov) [reply src/merge/mod.rs]: Unfortunately, https://doc.rust-lang.org/std/slice/trait.SliceConcatExt.html is not yet stable, and I see no nice way to put \n's _between_ the items. Maybe you know one?

Also, this is highly readable.

File: src/merge/mod.rs (hunk 2 above)
Comment (kyegupov) [reply src/merge/mod.rs]: done

File: src/merge/mod.rs (hunk 1 above)
Comment (maghoff) [reply src/merge/mod.rs]: In between! Of course :) Then I'd suggest using join, again to avoid introducing an extra dependency:

                .flat_map(...)
                .collect::<Vec<_>>()
                .join("\n")

Comment (kyegupov) [reply src/merge/mod.rs]: done (although you still pull in itertools anyway via bart_derive)

**Comment by maghoff** [comment]
Super, thanks.

I'd like to credit you in [CONTRIBUTORS.md](https://github.com/maghoff/sausagewiki/blob/master/CONTRIBUTORS.md). Should I write the credit to "kyegupov" or something else?

**Comment by kyegupov** [comment]
Konstantin Yegupov kyegupov4@gmail.com

**Comment by maghoff** [comment]
:heavy_check_mark: https://github.com/maghoff/sausagewiki/commit/bfbc81ada26ff1dc041e92625a963834920aac9c
